src/Data/MIME.hs

```diff
@@ -142,16 +231,33 @@ caseInsensitive = iso CI.mk CI.original
 {-# INLINE caseInsensitive #-}
 
 
-data ContentType = ContentType
-  (CI B.ByteString) -- type
-  (CI B.ByteString) -- subtype
-  Parameters
+-- | Content-Type header (RFC 2183).
+-- Use 'parameters' to access the parameters.
+-- Example:
+--
+-- @
+-- ContentType "text" "plain" [("charset", "utf-8")]
+-- @
+--
+data ContentType = ContentType (CI B.ByteString) (CI B.ByteString) Parameters
   deriving (Show)
 
+-- | Match content type.  If @Nothing@ is given for subtype, any
+-- subtype is accepted.
+--
+matchContentType
+  :: CI B.ByteString         -- ^ type
+  -> Maybe (CI B.ByteString) -- ^ optional subtype
+  -> ContentType
+  -> Bool
+matchContentType wantType wantSubtype (ContentType gotType gotSubtype _) =
+  wantType == gotType && maybe True (== gotSubtype) wantSubtype
+
 -- | Are the type and subtype the same? (parameters are ignored)
+--
 ctEq :: ContentType -> ContentType -> Bool
-ctEq (ContentType typ1 sub1 _) (ContentType typ2 sub2 _) =
-  typ1 == typ2 && sub1 == sub2
+ctEq (ContentType typ1 sub1 _) = matchContentType typ1 (Just sub1)
+{-# DEPRECATED ctEq "Use 'matchContentType' instead" #-}
```

Comment:
Is this actually necessary? The library isn't released yet, so no real reason as to deprecate it, rather then just deleting it?

Reply:
well... I'll delete it in a later commit, after I change purebred.  I just didn't want to make a breaking change in a doc commit.

Reply:
Then again, I probably shouldn't have made *any* code changes, but writing doc sometimes reveals bad APIs so it was a drive-by :)

Reply:
Thanks for reviewing @romanofski !